autopush/router/fcm.py

```diff
@@ -45,18 +45,19 @@ def amend_msg(self, msg, data=None):
             msg["senderid"] = data.get('creds', {}).get('senderID')
         return msg
 
-    def register(self, uaid, router_data, router_token=None, *kwargs):
+    def register(self, uaid, router_data, senderid=None, *kwargs):
         """Validate that the FCM Instance Token is in the ``router_data``"""
+        # "token" is the GCM registration id token generated by the client.
         if "token" not in router_data:
             raise self._error("connect info missing FCM Instance 'token'",
                               status=401)
-        # router_token and router_data['token'] are semi-legacy from when
-        # we were considering having multiple senderids for outbound
-        # GCM support. That was abandoned, but it is still useful to
-        # ensure that the client's senderid value matches what we need
-        # it to be. (If the client has an unexpected or invalid SenderID,
+        # senderid is the remote client's senderID value. This value is
+        # very difficult for the client to change, and there was a problem
+        # where some clients had an older, invalid senderID. We need to
+        # be able to match senderID to it's corresponding auth key.
+        # If the client has an unexpected or invalid SenderID,
         # it is impossible for us to reach them.
-        if not (router_token == router_data['token'] == self.senderID):
+        if not (senderid == self.senderID):
```

Comment:
router['token'] is the registrationID and should NOT match senderid. I'm an idiot.